Remove duplicates from array with slow calls

MarcinKasprowicz opened the pull request:
`CallsTable` was screaming to the console with following warning:
```
Warning: flattenChildren(...): Encountered two children with the same key
```
`props.items` array in `CallsTable` component had duplicates. Caused by: `nextState.items = this.state.items.concat(sortedItems);`

afronski commented on apps/xprof_gui/priv/app/call_tracer.jsx:
```diff
@@ -256,7 +255,12 @@ export default class CallsTracer extends React.Component {
         const lastId = sortedItems.length === 0 ? this.state.offset : _.last(sortedItems).id;
         nextState.offset = lastId;
-        nextState.items = this.state.items.concat(sortedItems);
+        let itemsToAdd = sortedItems.filter(item => {
```
Don't we have `lodash` there for a `.sort(...).uniq()` combo?

MarcinKasprowicz replied:
Sorry for laziness, my change is completely unreadable ... I will rewrite this :)

MarcinKasprowicz commented:
I did small refactor in CallsTracer component. Besides change in handling of items incoming from the backend I got rid of it all unnecessary re-renders. Now CallsTracer component will re-render when it really should, to confim that please build the frontend on you local env ale play with it to confirm

afronski commented on apps/xprof_gui/priv/package.json:
```diff
@@ -22,6 +22,7 @@
     "bootswatch": "3.3.7",
     "c3": "0.4.11",
     "jquery": "2.1.4",
+    "lodash": "^4.17.4",
```
TBH: I'd pin it to exact version - just to be safe. :wink:

afronski replied:
And that's the reason why `package.json` was reformated - I think you've used `npm install --save`. :wink:

MarcinKasprowicz replied:
Do want to lock only version of xprof's dependencies or also dependencies used by packages? Sticking to the exact version in package.json don't cover as from discrepancies. if we want so we should include package-lock.json in the repo. With it we will have consistent <b>node_module</b> tree. Exact versioning in package.json and use of package-lock.json will prevent us from that (npm >= 5 required)

gomoripeti replied:
including package-lock file sounds like a good idea. we also have rebar.lock for erlang dependencies already commited.

gomoripeti replied:
including `package-lock` sounds like a good idea. `rebar.lock` for erlang dependencies is already committed.

MarcinKasprowicz commented:
I have added `package-lock.json`. `package.json` is reformatted whenever I run `npm install` so I will keep it reformatted.

MarcinKasprowicz commented:
I replaced underscore with lodash, I didn't notice any problems in tests/app. I also removed functionality of START/STOP button, now it don't call capture `capture_data` endpoint.
